frame/dapps-staking/src/pallet/mod.rs

```diff
@@ -1259,5 +1342,24 @@ pub mod pallet {
                 Zero::zero()
             }
         }
+
+        fn check_next_delegate_reward(
+            staker: &T::AccountId,
+            contract_id: &T::SmartContract,
+            iteration: u8,
+        ) -> T::AccountId {
+            // If there is a delegation with call recursive
+            if DelegateRewardAccounts::<T>::contains_key(&staker, contract_id) && iteration < 3 {
```

Comment:
In general, loops/recursions should be avoided in runtime logic when possible. However, it's good you bounded it. How would you make the depth configurable?

How would you benchmark this solution?

And is it possible to implement a more simple solution?
E.g. if **Alice** delegates to **Bob**, and **Bob** delegates those rewards to **Charlie**, do we need to keep the original mapping of **Alice** to **Bob**?

Reply:
1. We can add a StorageValue. It would be great to define a maximum value (maybe no more than 5). But we would have to read the storage each time.
2. We can benchmark using the maximum possible value. 
3. I would not recommend doing that. Because if Bob unset the delegation to Charlie. You would lose the Alice to Bob record.